PowerShellTools/DebugEngine/ScriptDebugger.cs

```diff
@@ -219,7 +236,11 @@ public void UpdateBreakpoint(DebuggerBreakpointUpdatedEventArgs e)
         /// </summary>
         public void DebuggerFinished()
         {
-            IsDebuggingCommandReady = false;
+            lock (_debuggingCmdLock)
+            {
+                IsDebuggingCommandReady = false;

```

Comment:
Root cause of the race condition?  I think I'm missing something as far as what was being hit X-thread, and is fixed up with the lock around IsDebuggingCommandReady